### ISSUE TITLE
[Accessibility] Narrator does not read the image description of the existed image in the DataGridViewImageColumn

### ISSUE THREAD

**Issue by Jessie-Zhang01** [dotnet/winforms#3658]
* .NET Core Version: 
.Net SDK 5.0.100-rc.1.20378.13

* Have you experienced this same bug with .NET Framework?: 
Yes

More Info:
There is a similar bugs: [1159791 ](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1159791/?triage=true)in VSO.

**Minimal repro:**

1. Extract “[DataGridViewImage.zip](https://github.com/dotnet/winforms/files/4993008/DataGridViewImage.zip)” and open it in VS.
2. Build and run the project.
3. Open Narrator tool, navigate to the first row, Column2, check whether the narrator reads the image description of the image present.

**Expected behavior:**
Narrator should read the image description of the existed image in the DataGridViewImageColumn.

**Actual behavior:**
Narrator does not read the image description of the existed image in the DataGridViewImageColumn.
![image](https://user-images.githubusercontent.com/45864649/88764820-31e4a380-d1a8-11ea-8281-aa98999cce0e.png)

**Comment by Ryuugamine** [comment]
@merriemcgaw 
Could you please give me an example of how an image cell description should look?

**Comment by merriemcgaw** [comment]
@Ryuugamine - I'd say something like "picture of clouds" or at least the filename (not full path). Do we have any way currently to do that?

**Comment by merriemcgaw** [comment]
@guybark if the developer bound AccessibleDescription of the cell to some descriptive text in their database, would that be sufficient? @SergeySmirnov-Akvelon, would Narrator pick that up, or is there something we'd need to do so that it was read?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw.  I have investigated this issue and found that we use Description of DataGridViewImageCell as value for LegacyIAccessible.Description. Unfortunately Narrator doesn't announce LegacyIAccessible.Description and I see following possible ways for fixing it:
1. Add call of extra message for Narrator that will contain the description of image. For example next image:
![3658](https://user-images.githubusercontent.com/23376742/93207061-0bcc9000-f763-11ea-86b9-a953ab5e296f.jpg)
2. Update template of name for DataGridViewImageCell. Currently name contains address of cell and information about sorting if sorting is enabled (TestColumn Row 0, Sorted ascending). We can add data from description to this name. For example "TestColumn Row 0, Sorted ascending Test Image" or "TestColumn Row 0 Test Image, Sorted ascending".

Could you please share your opinion about it?

**Comment by merriemcgaw** [comment]
@SergeySmirnov-Akvelon  how does the description show up (or does it) in the UIA tree?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I have attached screenshot from Inspector:
![3658 - Inspector](https://user-images.githubusercontent.com/23376742/93307749-0413f680-f80a-11ea-9e19-7c83fb3e7f2b.jpg)

**Comment by merriemcgaw** [comment]
This looks reasonable to me. We've got the data in a reasonable property in the UIA tree. @guybark, what are your thoughts on reading/not reading the LegacyIAccessible.Description property? Does this look like a good UIA tree to you?

**Comment by merriemcgaw** [comment]
@Jessie-Zhang01 can you test this out with NVDA?

**Comment by Jessie-Zhang01** [comment]
@merriemcgaw verified this issue with .Net SDK 5.0.100-rc.2.20471.16, NVDA does not read the image description of the existed image in the DataGridViewImageColumn.
![image](https://user-images.githubusercontent.com/45864649/93861401-076c1e00-fcf3-11ea-9be8-a346236a1d1d.png)

**Comment by SergeySmirnov-Akvelon** [comment]
@merriemcgaw, what should we do in scope of this task?

**Comment by merriemcgaw** [comment]
assigning this to myself for now to make sure I have it on my list to discuss with @guybark.

**Comment by merriemcgaw** [comment]
Try using a table in Word or Outlook and seeing if an image in the table cell is announced and what property they are using to make it announce. If no one else does this I'd say we're done. We've reached out to Guy for more guidance.

**Comment by merriemcgaw** [comment]
It might be good to paste in the UIA Tree as it shows up in AI4Win and checking NVDA to see what they're doing as well.

**Comment by SergeySmirnov-Akvelon** [comment]
@merriemcgaw, I investigated it and found that images in Office and Outlook have no description. I have not found a way to add a description for a picture in these applications. Also I not found how set Description in WPF. The only thing that I was able to find out is that in the IE for the picture, it announces the following text: `Test, image, Test Description, selected`. Unfortunately, "Test Description" value is set for the "HelpText" and "Description" properties at the same time, so I can't tell exactly which one Narrator uses.

**Comment by merriemcgaw** [comment]
I've talked with our internal SME and he doesn't feel like we have anything more we need to do here. As long as we can set the UIA Value property for the cell we are in good shape. Please make sure that we are doing that in the sample app and we can close it! @SergeySmirnov-Akvelon

**Comment by SergeySmirnov-Akvelon** [comment]
User can use next code for setting of description for image:
```
private void Form1_Load(object sender, EventArgs e)
{
    // Add row with image to DataGriView
    dataGridView1.Rows.Add("Test", File.ReadAllBytes("Testing.jpg"));

    // Set description image in first row
    ((DataGridViewImageCell)dataGridView1.Rows[0].Cells[1]).Description = "Test Description";
}
```

**Comment by SergeySmirnov-Akvelon** [comment]
User can use next code for setting of description for image:
```
private void Form1_Load(object sender, EventArgs e)
{
    // Add row with image to DataGriView
    dataGridView1.Rows.Add("Test", File.ReadAllBytes("Testing.jpg"));

    // Set description for image in first row
    ((DataGridViewImageCell)dataGridView1.Rows[0].Cells[1]).Description = "Test Description";
}
```
![Issue-3658](https://user-images.githubusercontent.com/23376742/97277793-fbbbcc00-1849-11eb-9bbb-a444d15e6553.png)

I attached [test project](https://github.com/dotnet/winforms/files/5444004/DataGridViewImage.zip) with example of this code.

**Comment by merriemcgaw** [comment]
@SergeySmirnov-Akvelon can you get this added to the Accessibility test app to show best practices and then we can close this issue again?

**Comment by SergeySmirnov-Akvelon** [comment]
> I attached [test project](https://github.com/dotnet/winforms/files/5444004/DataGridViewImage.zip) with example of this code.

@merriemcgaw, test project already is added. Could you please add this link to this issue description?

**Comment by merriemcgaw** [comment]
I think we're good to close this issue. I'll confirm if the compliance testers have verified the issue.
